internal/dataplane/translator/subtranslator/httproute.go

```diff
@@ -1102,6 +1122,30 @@ func generateKongRouteModifierFromExtensionRef(pluginNamesFromExtensionRef []str
 	}
 }
 
+// hostPortFromHTTPPathModifier returns the part of scheme and hostport in the URL of `Location` header
+// for generating configuration of Kong plugins to satisfy `requestRedirect` filter of HTTPRoute.
+func schemeHostPortFromHTTPPathModifier(modifier *gatewayapi.HTTPRequestRedirectFilter) (string, string) {
+	// Since the `redirect` plugin requires the scheme, we do not change the logic to use `http` scheme when scheme is empty.
+	// TODO: According to the spec of gateway API, the original scheme should be kept if no scheme is given.
+	// So the logic may be changed if Kong gateway supports.
+	scheme := "http"
+	if modifier.Scheme != nil {
+		scheme = *modifier.Scheme
+	}
+	hostname := ""
+	if modifier.Hostname != nil {
+		hostname = string(*modifier.Hostname)
+	}
+	// As gateway API specified, if `Port` is nil, port should be left empty for using well-known port for the scheme.
+	// REVIEW: should we keep the default port 80 as the existing implementation?
```

Comment:
Technically it will break users who uses `https` scheme and empty port and wanted to redirect the request to port `80` with scheme `https`. But I think redirecting `https` requests to port 80 is not a common usage.

Reply:
Updated the comments here. @pmalek do you think we should mention it in changlog/docs and mark it as a breaking change?

Reply:
Yes, I'd add a note in Changelog mentioning that this is done to achieve conformance with Gateway API.

Not sure about mentioning this in the breaking changes section, but I'd add a note about "possibly breaking some uncommon use cases". WDYT?